fontbe/src/orchestration.rs

```diff
@@ -219,8 +220,99 @@ impl From<&VariationRegion> for TupleBuilder {
     }
 }
 
+#[repr(u8)]
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum LocaFormat {
+    Short,
+    Long,
+}
+
+impl LocaFormat {
+    pub fn new(loca: &[u32]) -> LocaFormat {
+        // https://github.com/fonttools/fonttools/blob/1c283756a5e39d69459eea80ed12792adc4922dd/Lib/fontTools/ttLib/tables/_l_o_c_a.py#L37
+        if loca.last().copied().unwrap_or_default() < 0x20000
+            && loca.iter().all(|offset| offset % 2 == 0)
+        {
+            LocaFormat::Short
+        } else {
+            LocaFormat::Long
+        }
+    }
+}
+
+impl TryFrom<u8> for LocaFormat {
+    type Error = ();
+
+    fn try_from(value: u8) -> Result<Self, ()> {
+        match value {
+            _ if value == LocaFormat::Short as u8 => Ok(LocaFormat::Short),
+            _ if value == LocaFormat::Long as u8 => Ok(LocaFormat::Short),
+            _ => Err(()),
+        }
+    }
+}
+
+// Free function of specific form to fit macro
+fn loca_format_from_file(file: &Path) -> LocaFormat {
+    let bytes = fs::read(file).unwrap();
+    LocaFormat::try_from(*bytes.first().unwrap()).unwrap()
+}
+
+// Free function of specific form to fit macro
+fn loca_format_to_bytes(format: &LocaFormat) -> Vec<u8> {
+    vec![*format as u8]
+}
+
 pub type BeWork = dyn Work<Context, Error> + Send;
-type GlyfLoca = (Vec<u8>, Vec<u32>);
+pub struct GlyfLoca {
+    pub glyf: Vec<u8>,
+    pub loca: Vec<u32>,
+}
+
+impl GlyfLoca {
+    pub fn new(glyf: Vec<u8>, loca: Vec<u32>) -> Self {
+        Self { glyf, loca }
+    }
+
+    pub fn read(format: LocaFormat, paths: &Paths) -> Self {
+        let glyf = read_entire_file(&paths.target_file(&WorkId::Glyf));
+        let raw_loca = read_entire_file(&paths.target_file(&WorkId::Loca));
+        let loca = if format == LocaFormat::Short {
+            raw_loca[1..]
+                .chunks_exact(std::mem::size_of::<u16>())
+                .map(|bytes| u16::from_be_bytes(bytes.try_into().unwrap()) as u32 * 2)
+                .collect()
+        } else {
+            raw_loca[1..]
```

Comment:
should we still be skipping the first byte, here? is there a test that would catch this?

Reply:
We should not, that's a bug leftover from my initial hackery. Evidently there is no such test, will add.